### DISCUSSION TITLE
locking the dock contents

### DISCUSSION THREAD

**Discussion by frank3427** [geerlingguy/ansible-collection-mac#15] [Q&A]
have you been able to lock the content of the dock,  I want to setup a few iMac's a with a defined  order of apps on the dock but want to restrict users from being able to add or remove items from the doc.

according to many guides the following should work but does not

`defaults write com.apple.Dock contents-immutable -bool yes; killall Dock

You'll now notice that you can't rearrange the position of apps on the Dock, add new apps by dragging them to the Dock from Finder, or use the right-click "Keep in Dock" option on a running application. You also won't be able to remove apps or folders from the Dock.`

**Comment by geerlingguy** [comment]
Some things that worked in past OS versions don't anymore... I've found esoteric options like locking the dock are not well-documented and prone to breaking or changing plist values in future versions of macOS.

I typically just manage all the dock items with my Dock role, so they are 'immutable' in the fact that the Dock is reset back to how I like it every time I run the playbook.
